src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/ProjectSystem/Rules/XamlRuleTests.cs

```diff
@@ -0,0 +1,289 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using Xunit;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Rules
+{
+    public class XamlRuleTests
+    {
+        [Theory]
+        [MemberData(nameof(GetFileItemRules))]
+        public void FileRulesShouldMatchNone(string ruleName, string fullPath)
+        {
+            // Special case for Folder rule which hasn't been split yet, but is in the Items folder. But also its completely different.
+            if (ruleName.Equals("Folder", StringComparison.Ordinal))
+            {
+                return;
+            }
+            // No need to check None against None
+            if (ruleName.Equals("None", StringComparison.Ordinal))
+            {
+                return;
+            }
+
+            string noneFile = Path.Combine(fullPath, "..", "None.xaml");
+
+            XmlDocument none = LoadXamlRule(noneFile);
+            XmlDocument rule = LoadXamlRule(fullPath);
+
+            // First fix up the Name as we know they'll differ.
+            rule.DocumentElement.Attributes["Name"].Value = "None";
+
+            AssertXmlEqual(none, rule);
+        }
+
+        [Theory]
+        [MemberData(nameof(GetBrowseObjectItemRules))]
+        public void BrowseObjectRulesShouldMatchNone(string ruleName, string fullPath)
+        {
+            // Special case for Folder rule which hasn't been split yet, but is in the Items folder. But also its completely different.
+            if (ruleName.Equals("Folder", StringComparison.Ordinal))
+            {
+                return;
+            }
+            // No need to check None against None
+            if (ruleName.Equals("None", StringComparison.Ordinal))
+            {
+                return;
+            }
+
+            string noneFile = Path.Combine(fullPath, "..", "None.BrowseObject.xaml");
+
+            XmlDocument none = LoadXamlRule(noneFile);
+            XmlDocument rule = LoadXamlRule(fullPath);
+
+            // First fix up the Name and DisplayName as we know they'll differ.
+            rule.DocumentElement.Attributes["Name"].Value = "None";
+            rule.DocumentElement.Attributes["DisplayName"].Value = "General";
+
+            AssertXmlEqual(none, rule);
+        }
+
+        [Theory]
+        [MemberData(nameof(GetAllRules))]
+        public void NonVisiblePropertiesShouldntBeLocalized(string ruleName, string fullPath)
+        {
+            XmlDocument rule = LoadXamlRule(fullPath);
+
+            foreach (XmlNode node in rule.DocumentElement.ChildNodes)
+            {
+                if (node is XmlElement element && element.HasAttribute("Visible") && element.Attributes["Visible"].Value.Equals("false", StringComparison.OrdinalIgnoreCase))
```

Comment:
Is a property that omits the `Visible` attribute considered visible?

Reply:
Yes, the default is true.

https://github.com/Microsoft/msbuild/blob/e70a3159d64f9ed6ec3b60253ef863fa883a99b1/src/Framework/XamlTypes/BaseProperty.cs#L52